src/shared/extensionUtilities.ts

```diff
@@ -109,6 +132,24 @@ export function isCloud9(flavor: 'classic' | 'codecatalyst' | 'any' = 'any'): bo
     return (flavor === 'classic' && !codecat) || (flavor === 'codecatalyst' && codecat)
 }
 
+export function isSageMaker(): boolean {
+    return vscode.env.appName === sageMakerAppname
+}
+
+/**
+ * Helper function to determine if Cloud9 or Sagemaker are being used. Both use IAM roles
+ * on hosting instance for auth
```

Comment:
- describe the semantics not the implementation.
- "helper" is never needed in a description or name. 

```suggestion
 * Returns true if credentials are provided by the environment (typically via ~/.aws/).
```

Reply:
updated